mongodb-datastore/db/mongodb_connection.go

```diff
@@ -50,35 +55,40 @@ func (m *MongoDBConnection) EnsureDBConnection() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	if m.client == nil {
-		fmt.Println("No MongoDB client has been initialized yet. Creating a new one.")
+		logger.Info("No MongoDB client has been initialized yet. Creating a new one.")
 		return m.connectMongoDBClient()
 	} else if err = m.client.Ping(ctx, nil); err != nil {
-		fmt.Println("MongoDB client lost connection. Attempt reconnect.")
+		logger.Info("MongoDB client lost connection. Attempt reconnect.")
```

Comment:
For consistency reasons, I would reword this to "Attempting reconnect"
Like above with "Creating a new one"

Reply:
sure, will edit in all PRs, thx :)